File: frontend/notice-web/src/types/models/sync-conflict.ts
import type { ConflictStatus, ConflictType, SyncEntityType } from '@/types/enums/sync'
import type { DatabaseType } from '@/types/enums/db'
import type { SyncConflictItem } from './sync-conflict-item'

export interface SyncConflictListVo {
  id: string
  entityType: SyncEntityType
  entityId: string
  status: ConflictStatus
  conflictType: ConflictType
  firstSeenAt?: string
  lastSeenAt?: string
  lastCheckedAt?: string
  lastNotifiedAt?: string
  notifyCount?: number
  resolutionSourceDb?: DatabaseType
  resolutionNote?: string
  resolvedAt?: string
  items?: SyncConflictItem[]
}

export interface SyncConflictDetailVo {
  id: string
  entityType: SyncEntityType
  entityId: string
  status: ConflictStatus
  conflictType: ConflictType
  firstSeenAt?: string
  lastSeenAt?: string
  lastCheckedAt?: string
  lastNotifiedAt?: string
  notifyCount?: number
  resolutionSourceDb?: DatabaseType
  resolutionNote?: string
  resolvedAt?: string
  createTime?: string
  updateTime?: string
  items: SyncConflictItem[]
}

export interface SyncConflictQuery {
  pageNo?: number
  pageSize?: number
  beginTime?: string
  endTime?: string
  entityType?: SyncEntityType
  status?: ConflictStatus | string
  conflictType?: ConflictType | string
  sourceDb?: DatabaseType
  limitToOpenOnly?: boolean
}

export interface SyncConflictResolvePayload {
  sourceDb: DatabaseType
  note?: string
}

File: backend/src/main/java/com/notice/system/entity/SyncConflict.java
package com.notice.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.notice.system.entityEnum.ConflictStatus;
import com.notice.system.entityEnum.ConflictType;
import com.notice.system.entityEnum.DatabaseType;
import com.notice.system.entityEnum.SyncEntityType;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("sync_conflict")
public class SyncConflict {

    @TableId(type = IdType.ASSIGN_UUID)
    private String id;

    @TableField(value = "entity_type")
    private SyncEntityType entityType;

    @TableField(value = "entity_id")
    private String entityId;

    private ConflictStatus status;
    private ConflictType conflictType;

    private LocalDateTime firstSeenAt;
    private LocalDateTime lastSeenAt;
    private LocalDateTime lastCheckedAt;

    private LocalDateTime lastNotifiedAt;
    private Integer notifyCount;

    private DatabaseType resolutionSourceDb;
    private String resolutionNote;
    private LocalDateTime resolvedAt;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}


